### DISCUSSION TITLE
How to make a controlled input, when using a controller?

### DISCUSSION THREAD

**Discussion by sabbiu** [react-hook-form/react-hook-form#3979] [General] [answered]
Follow these steps to understand what I am trying to achieve.

- Open [code sandbox demo](https://codesandbox.io/s/react-hook-form-controller-forked-zy6zy?file=/src/index.js)
- All three inputs (switch, checkbox and input) has same name property.
- Check the Input Labelled `MUI Checkbox`
- You can observe its value reflected down in input field but not in switch.

How can I make sure for it to be reflected in both of those fields?

**Comment by bluebill1049** [comment]
checkbox is `checked` right? not `value`

**Reply by sabbiu** [reply]
I changed it to `checked`. But still, I am having the same issue. [Updated codesandbox](https://codesandbox.io/s/react-hook-form-controller-forked-zy6zy?file=/src/index.js)

**Reply by bluebill1049** [reply]
did you see the example in the doc? https://react-hook-form.com/api#Controller

```
<Controller
  control={control}
  name="test"
  render={(
    { onChange, onBlur, value, name, ref },
    { invalid, isTouched, isDirty }
  ) => (
    <Checkbox
      onBlur={onBlur}
      onChange={(e) => onChange(e.target.checked)}
      checked={value}
      inputRef={ref}
    />
  )}
/>
```

<img width="723" alt="Screen Shot 2021-01-21 at 8 45 11 pm" src="https://user-images.githubusercontent.com/10513364/105333204-8f265280-5c29-11eb-9491-96211b780bba.png">

**Reply by sabbiu** [reply]
Yes, I saw that example. But, adding `inputRef` also didn't work. [Updated Codesandbox](https://codesandbox.io/s/react-hook-form-controller-forked-zy6zy?file=/src/index.js)

**Reply by bluebill1049** [reply]
what's not working?

**Reply by bluebill1049** [reply]
there is an example in the doc as well: https://codesandbox.io/s/react-hook-form-v6-controller-qsd8r?file=/src/Mui.js

**Reply by sabbiu** [reply]
When I turn on the `MUI Checkbox`, I want `MUI Switch` also to turn on. As they both share the same `name`.

**Reply by bluebill1049** [reply]
> Important: name is required and unique (can not start with a number or use number as key name). Input name also supports dot and bracket syntax, which allows you to easily create nested form fields.

https://react-hook-form.com/api#register

read the doc, name should be unique.

**Reply by sabbiu** [reply]
Totally missed that out. BTW, I was under the impression that we can create multiple components with the same name, and the same value gets reflected in all of them.

Thanks for your help

**Comment by sabbiu** [comment, marked as answer]
@bluebill1049  I was able to create a hack, to fulfill my use case. IDK, if this is performant solution or not. If you can take a look into it, it would be awesome. Thanks
[Codesandbox Link](https://codesandbox.io/s/react-hook-form-controller-forked-zy6zy?file=/src/index.js)

**Reply by sabbiu** [reply]
This is what I wanted to achieve.

**Before**
![rhf-before](https://user-images.githubusercontent.com/19733021/105344909-ace4bf00-5c0b-11eb-83c8-5f0de2e50f74.gif)

**After**
![rhf-after](https://user-images.githubusercontent.com/19733021/105344924-b2420980-5c0b-11eb-9ae5-07070914af25.gif)

**Reply by bluebill1049** [reply]
i would use `useWatch` instead

**Reply by sabbiu** [reply]
Thanks! It works. Here is the final solution, [codesandbox](https://codesandbox.io/s/react-hook-form-controller-forked-uns8u?file=/src/index.js)
